Expand `$args` array parameters that are described in other functions

tellyworth opened the pull request:
Quick WIP version. See #95.

Before, on the get_terms() function reference:
<img width="1092" alt="Screen Shot 2022-06-08 at 4 13 40 pm" src="https://user-images.githubusercontent.com/7200686/172544761-491de5bf-f0d3-44a6-b3db-40bb97caaf01.png">

After, collapsed (detail/summary tag - note the teeny tiny triangle):
<img width="1011" alt="Screen Shot 2022-06-08 at 4 15 11 pm" src="https://user-images.githubusercontent.com/7200686/172545485-4bbc0c96-36a5-41ab-b4c2-673a8bc7d759.png">

After, expanded:
<img width="1005" alt="Screen Shot 2022-06-08 at 4 15 25 pm" src="https://user-images.githubusercontent.com/7200686/172545535-ef2d99a6-03f9-4aa7-b4b2-506d0c3b8874.png">

Before, on get_posts(). Note that this one has a few arguments explicitly documented:
<img width="979" alt="Screen Shot 2022-06-08 at 4 21 35 pm" src="https://user-images.githubusercontent.com/7200686/172546064-a6b32bc1-8c5d-4998-8d88-c5f03f6e9e7c.png">

After, collapsed:
<img width="965" alt="Screen Shot 2022-06-08 at 4 21 51 pm" src="https://user-images.githubusercontent.com/7200686/172546227-e502eba8-427b-49ad-ad22-3e2ffcae0b4f.png">

After, expanded (note that it still has the explicit docs first, I haven't hidden that):
<img width="988" alt="Screen Shot 2022-06-08 at 4 22 28 pm" src="https://user-images.githubusercontent.com/7200686/172546334-1b02b7a3-4d27-4626-98ed-d7cbc1d3426a.png">

There are definitely some improvements needed especially on the front-end. But this seems to work pretty well at recursing through and finding the correct array. For example it works with multiple levels of indirection, like `wp_count_terms($args) -> get_terms($args) -> WP_Term_Query::__construct($query)`.

tellyworth committed:
Expand `$args` array parameters that are described in other functions

Quick WIP version.

tellyworth committed:
Remove cruft

tellyworth committed:
Fix for wp_update_post which uses `$postarr`

ryelle committed:
Fix summary & description semantics

ryelle commented:
I pushed up a commit to remove the `summary` wrapper from the description — that was stripping the semantics from everything inside it. It's best to just have the content in `summary` be a single text node (or heading).

I just changed it so it's "More Arguments". Feel free to change that to whatever works better :)

An example on `get_posts`:

<img width="994" alt="Screen Shot 2022-06-08 at 4 26 53 PM" src="https://user-images.githubusercontent.com/541093/172711210-9e938f53-da28-4f57-8463-17ecebe70de1.png">

And on `wp_count_terms`:

<img width="990" alt="Screen Shot 2022-06-08 at 4 27 03 PM" src="https://user-images.githubusercontent.com/541093/172711214-18146a9a-e5af-4675-91ea-0438d92ed65e.png">

I'm also seeing `<br>` on all the params now, I think that's something new on this branch.

tellyworth committed:
Make the summary look like a link

I'm sure this is the wrong way to solve the problem, but at least it looks more clickable.

tellyworth committed:
Use a more explicit summary "More Arguments from the_func( ...$arg)"

tellyworth commented:
Thanks!

I'm not seeing the `<br>` thing at all. Not sure what's up with that. Some kind of nl2br formatting, maybe on import? I re-ran an import and it didn't reproduce.

I pushed a couple of commits riffing on this a bit. The summary now looks more obviously clickable, and explains where the extra info came from:
<img width="954" alt="Screen Shot 2022-06-09 at 5 12 44 pm" src="https://user-images.githubusercontent.com/7200686/172787270-b9c7f349-cbbb-425b-abc7-090d398835eb.png">

It's a bit better but still needs some work. Might be almost good enough to merge now for a designer to look at later - what do you think? Assuming I can fix the br thing.

ryelle reviewed (approved):
> I'm not seeing the <br> thing at all. Not sure what's up with that. Some kind of nl2br formatting, maybe on import? I re-ran an import and it didn't reproduce.

🤷🏻  I'm not seeing them on trunk, just this branch. Ah okay — I just rebased and they're gone now. I'm not sure why you weren't seeing them but when this merges it should be fine, at least.

> Might be almost good enough to merge now for a designer to look at later - what do you think?

That works for me. I don't think it should look like a link (since it's not taking you anywhere), but we can iterate on that later.

tellyworth committed:
Fix rebase error
